libraries/standard/http/test/test-HTTPClient_AddHeader.c

```diff
@@ -1,9 +1,182 @@
+      HTTP_TEST_HEADER_VALUE_LEN +        \
+      HTTP_HEADER_LINE_SEPARATOR_LEN +    \
+      HTTP_HEADER_LINE_SEPARATOR_LEN )
+
 int main()
 {
-    return 0;
-}
\ No newline at end of file
+    HTTPRequestHeaders_t reqHeaders = HTTP_REQUEST_HEADERS_INITIALIZER;
+    HTTPRequestHeaders_t reqHeadersDflt = HTTP_REQUEST_HEADERS_INITIALIZER;
+    HTTPStatus_t test_err = HTTP_NOT_SUPPORTED;
+    uint8_t buffer[ HTTP_TEST_BUFFER_SIZE ] = { 0 };
+    uint8_t smallBuffer[ HTTP_TEST_SUFFICIENT_HEADER_LEN - 1 ] = { 0 };
+    char correctHeader[ HTTP_TEST_BUFFER_SIZE ] = { 0 };
+    size_t correctHeaderLen = HTTP_TEST_SUFFICIENT_HEADER_LEN;
+    struct Header
+    {
+        char field[ 100 ];
+        size_t fieldLen;
+        char value[ 100 ];
+        size_t valueLen;
+    }
+    header;
+
+/* Write template header field and value to a struct to pass as
+ * parameters to HTTPClient_AddHeader() method. */
+#define fillHeaderStructTemplate()                                    \
+    do {                                                              \
+        memcpy( header.field,                                         \
+                HTTP_TEST_HEADER_FIELD, HTTP_TEST_HEADER_FIELD_LEN ); \
+        header.fieldLen = HTTP_TEST_HEADER_FIELD_LEN;                 \
+        memcpy( header.value,                                         \
+                HTTP_TEST_HEADER_VALUE, HTTP_TEST_HEADER_VALUE_LEN ); \
+        header.valueLen = HTTP_TEST_HEADER_VALUE_LEN;                 \
+    }                                                                 \
+    while( 0 )
+
+#define reset()                                                      \
+    do {                                                             \
+        test_err = HTTP_NOT_SUPPORTED;                               \
+        reqHeaders = reqHeadersDflt;                                 \
+        memset( buffer, 0, HTTP_TEST_BUFFER_SIZE );                  \
+        memset( correctHeader, 0, HTTP_TEST_SUFFICIENT_HEADER_LEN ); \
+        memset( header.field, 0, 100 );                              \
+        header.fieldLen = 0;                                         \
+        memset( header.value, 0, 100 );                              \
+        header.valueLen = 0;                                         \
+    }                                                                \
+    while( 0 )
+
+    plan( 16 );
+
+    /* Test the happy path. */
+    reset();
+    fillHeaderStructTemplate();
+    /* We add 1 because snprintf() writes a null byte at the end. */
+    snprintf( correctHeader, HTTP_TEST_SUFFICIENT_HEADER_LEN + 1,
+              "%s%s: %s\r\n\r\n",
+              HTTP_TEST_HEADER_REQUEST_LINE,
+              HTTP_TEST_HEADER_FIELD, HTTP_TEST_HEADER_VALUE );
+    correctHeaderLen = HTTP_TEST_SUFFICIENT_HEADER_LEN;
+    /* Set parameters for reqHeaders. */
+    memcpy( buffer, HTTP_TEST_HEADER_REQUEST_LINE, HTTP_TEST_HEADER_REQUEST_LINE_LEN );
```

Comment:
How do you know that buffer is big enough? It's declared independently of the defines that determine what's written into it.